ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -106,16 +108,31 @@ public void run(MLAgent mlAgent, Map<String, String> params, ActionListener<Obje
         List<MLToolSpec> toolSpecs = mlAgent.getTools();
         String memoryType = mlAgent.getMemory().getType();
         String memoryId = params.get(MEMORY_ID);
+        // for regenerate, original interaction id must provide
+        String regenerateInteractionId = params.get(REGENERATE_INTERACTION_ID);
```

Comment:
Can interaction regeneration logic be handled in MLAgenExecutor instead of Chat agent? In future there could be new agent types and we should not re-implement regeneration logic in each agent type

Reply:
@arjunkumargiri Instead of put logic in `MLAgenExecutor`, we can have a base class for all kinds of agent runner, put common logic in that class would make more sense.

Reply:
Agreed, makes sense.

Reply:
@arjunkumargiri Since the root interaction been moved to `MLAgentRunner`,regenerate is coupled with root interaction, move the logic to that class also.